// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/models/AwsRegions.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pineonce.io Public API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AwsRegionsToJSON = exports.AwsRegionsFromJSONTyped = exports.AwsRegionsFromJSON = exports.AwsRegions = void 0;
/**
 * Regions available with the AWS cloud provider
 * @export
 */
exports.AwsRegions = {
    East1: 'us-east-1',
    West1: 'us-west-1',
    West2: 'us-west-2'
};
function AwsRegionsFromJSON(json) {
    return AwsRegionsFromJSONTyped(json, false);
}
exports.AwsRegionsFromJSON = AwsRegionsFromJSON;
function AwsRegionsFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.AwsRegionsFromJSONTyped = AwsRegionsFromJSONTyped;
function AwsRegionsToJSON(value) {
    return value;
}
exports.AwsRegionsToJSON = AwsRegionsToJSON;
