// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/models/GcpRegions.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pineonce.io Public API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GcpRegionsToJSON = exports.GcpRegionsFromJSONTyped = exports.GcpRegionsFromJSON = exports.GcpRegions = void 0;
/**
 * Regions available with the GCP cloud provider
 * @export
 */
exports.GcpRegions = {
    UsWest1: 'us-west1',
    UsWest2: 'us-west2',
    EuWest4: 'eu-west4',
    NorthamericaNortheast1: 'northamerica-northeast1',
    AsiaNortheast1: 'asia-northeast1',
    AsiaSoutheast1C: 'asia-southeast1C'
};
function GcpRegionsFromJSON(json) {
    return GcpRegionsFromJSONTyped(json, false);
}
exports.GcpRegionsFromJSON = GcpRegionsFromJSON;
function GcpRegionsFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.GcpRegionsFromJSONTyped = GcpRegionsFromJSONTyped;
function GcpRegionsToJSON(value) {
    return value;
}
exports.GcpRegionsToJSON = GcpRegionsToJSON;
